File: credit-check-service/src/main/java/com/senacor/innolab/graalvm/integration/creditdetailservice/model/CreditDetails.java
package com.senacor.innolab.graalvm.integration.creditdetailservice.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import io.quarkus.runtime.annotations.RegisterForReflection;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@RegisterForReflection
@JsonIgnoreProperties(ignoreUnknown = true)
public class CreditDetails {

    private Long id;
    private BigDecimal amount;
    private LocalDate start;
    private LocalDate end;
    private BigDecimal interestRate;

}

File: credit-details-service/src/main/kotlin/com/senacor/innolab/graalvm/credit/details/model/CreditDetails.kt
package com.senacor.innolab.graalvm.credit.details.model

import com.fasterxml.jackson.annotation.JsonFormat
import java.math.BigDecimal
import java.time.LocalDate
import javax.persistence.*

@Entity
@Table(name = "credit")
class CreditDetails(

    @Id
    @Column(nullable = false)
    var id: String,

    @Column
    var amount: BigDecimal,

    @Column
    @JsonFormat(pattern = "yyyy-MM-dd")
    var start: LocalDate,

    @Column
    @JsonFormat(pattern = "yyyy-MM-dd")
    var end: LocalDate,

    @Column
    var interestRate: BigDecimal
) {

    companion object {
        fun of(id: String, amount: BigDecimal, startDate: String, endDate: String, interestRate: BigDecimal) =
            CreditDetails(
                id,
                amount,
                LocalDate.parse(startDate),
                LocalDate.parse(endDate),
                interestRate
            )
    }
}